examples/spring-boot-with-certmanager-example/README.md

```diff
@@ -0,0 +1,118 @@
+# Spring Boot with Cert-Manager
+
+The purpose of this Spring Boot example is to demonstrate how we can configure a HTTPS/TLS microservice using the Cert-Manager Dekorate extension.
+
+Apart from the Dekorate Kubernetes starter dependency, the application must declare the Cert-Manager dekorate dependency part of the pom.xml file:
+
+    <dependency>
+      <groupId>io.dekorate</groupId>
+      <artifactId>certmanager-annotations</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+
+## Enable HTTPS transport
+
+To enable the HTTPS/TLS transport in Spring Boot, we need to add the following properties:
+
+```
+server.port=8443
+server.ssl.enabled=true
+```
+
+Next, we need to configure the Java PKCS12 Keystore properties that our Spring Boot application will use to get the Server certificate signed and the private key:
+
+```
+server.ssl.key-store-type=PKCS12
+server.ssl.key-store=/path/to/keystore.p12
+server.ssl.key-store-password=the password
+```
+
+Where is the `keystore.p12` Keystore and what is its password? This is where Cert-Manager comes to play. This Java Keystore will be generated by Cert-Manager on the Kubernetes platform as a secret resource. Then, what we need to do is to mount a volume from this generated secret, so the Spring Boot application can read the generated Keystore by Cert-Manager. In the following sections, we'll see how we can instruct Cert-Manager to generate the Keystore and how to configure the application to mount the secret and use it.
+
+## Generate a Self-Signed Certificate and the Keystore
+
+Let's configure the different properties to request the generation of the Self-Signed certificate and the `keystore.p12` PKCS12 keystore file:
+
+```
+dekorate.certificate.secret-name=tls-secret
+dekorate.certificate.self-signed.enabled=true
+dekorate.certificate.keystores.pkcs12.create=true
+# the secret name of the password:
+dekorate.certificate.keystores.pkcs12.passwordSecretRef.name=pkcs12-pass 
+dekorate.certificate.keystores.pkcs12.passwordSecretRef.key=password
+```
+
+Using this configuration, Dekorate will create the `Certificate` and `Issuer` resources that, once installed on the Kubernetes platform, will be used by Certificate Manager to generate a self-signed certificate and the Keystore files within the secret `tls-secret`. 
+
+About the Keystore password, we have configured our certificate resource to get the password from other secret resource named `pkcs12-pass` with a data key `password` and value in base 64 is `supersecret`, so let's create this secret in the file `src/main/resources/k8s/common.yml`:
```

Comment:
"About the Keystore password, we have configured our certificate resource to get the password from other secret resource named `pkcs12-pass` with a data key `password` and value in base 64 is `supersecret`, so let's create this secret in the file `src/main/resources/k8s/common.yml`:"

-->

"**NOTE**: As the keystore file (pkcs12, ...) is password protected, this is then the reason why we have to create a secret including the needed password. For that purpose, we are going to create, part of the file `src/main/resources/k8s/common.yml`, a secret named "pkcs12-pass". The data field will include the key password where the string `supersecret` will be encoded in base64:"

Reply:
I don't think this should be a note, but a mandatory step. I will use the rest of the changes.